app/models/agents/post_agent.rb

```diff
@@ -162,22 +171,29 @@ def normalize_response_headers(headers)
       }
     end
 
-    def handle(data, payload = {}, headers)
-      url = interpolated(payload)[:post_url]
+    def handle(data, event = Event.new, headers)
```

Comment:
Sidenote: it's odd that we have a middle parameter with a default. I wonder if we ever use that?